src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala

```diff
@@ -434,6 +433,46 @@ object TaskGroupBuilder {
     builder
   }
 
+  /**
+    * Computes all port env vars for the given container.
+    * Form: PORT{INDEX}=123
+    */
+  private[this] def containerPortEnvVars(
+    pod: PodDefinition,
+    container: MesosContainer,
+    hostPorts: Seq[Option[Int]],
+    config: BuilderConfig): Map[String, String] = {
+    val hostNetworkMode = pod.networks.contains(HostNetwork)
+    val networkPorts = if (hostNetworkMode) hostPorts else pod.containers.flatMap(_.endpoints).map(_.containerPort)
+    val allEndpoints = pod.containers.flatMap(_.endpoints.map(_.name)).zip(networkPorts).toMap.withDefaultValue(None)
+    val containerEndpoints = allEndpoints.filter{ case (name, port) => container.endpoints.exists(_.name == name) }
+    portEnvVars(container.endpoints, containerEndpoints.values.toVector, config.envVarsPrefix)
+  }
+
+  /**
+    * Computes all endpoint env vars for the entire pod definition
+    * Form:
+    * ENDPOINT_NAMES={ENDPOINT_NAME},{ENDPOINT_NAME},..,{ENDPOINT_NAME}
```

Comment:
I think its sufficient to follow a single pattern, e.g. ENDPOINT_${NAME}=${HOST_PORT}